src/uu/split/src/filenames.rs

```diff
@@ -54,21 +67,200 @@ impl SuffixType {
+                }
+            }
+            (_, true, _, _) => {
+                stype = SuffixType::Hexadecimal;
+                // if option was specified, but without value - this will return None as there is no default value
+                if let Some(opt) = matches.get_one::<String>(OPT_HEX_SUFFIXES) {
+                    start = usize::from_str_radix(opt, 16)
+                        .map_err(|_| SuffixError::NotParsable(opt.to_string()))?;
+                    auto_widening = false;
+                }
+            }
+            (_, _, true, _) => stype = SuffixType::Decimal, // short numeric suffix '-d'
+            (_, _, _, true) => stype = SuffixType::Hexadecimal, // short hex suffix '-x'
+            _ => stype = SuffixType::Alphabetic, // no numeric/hex suffix, using default alphabetic
+        }
+
+        // Get suffix length and a flag to indicate if it was specified with command line option
+        let (mut length, is_length_cmd_opt) =
+            if let Some(v) = matches.get_one::<String>(OPT_SUFFIX_LENGTH) {
+                // suffix length was specified in command line
+                (
+                    v.parse::<usize>()
+                        .map_err(|_| SuffixError::NotParsable(v.to_string()))?,
+                    true,
+                )
+            } else {
+                // no suffix length option was specified in command line
+                // set to default value
+                (default_length, false)
+            };
+
+        // Disable dynamic auto-widening if suffix length was specified in command line with value > 0
+        if is_length_cmd_opt && length > 0 {
+            auto_widening = false;
+        }
+
+        // Auto pre-calculate new suffix length (auto-width) if necessary
+        if let Strategy::Number(ref number_type) = strategy {
+            let chunks = number_type.num_chunks();
+            let required_length = ((start as u64 + chunks) as f64)
+                .log(stype.radix() as f64)
+                .ceil() as usize;
+
+            if (start as u64) < chunks && !(is_length_cmd_opt && length > 0) {
+                // with auto-width ON the auto-widening is OFF
+                auto_widening = false;
+
+                // do not reduce suffix length with auto-width
+                if length < required_length {
+                    length = required_length;
+                }
+            }
+
+            if length < required_length {
+                return Err(SuffixError::TooSmall(required_length));
+            }
+        }
+
+        // Check edge case when suffix length == 0 was specified in command line
+        // Set it to default value
+        if is_length_cmd_opt && length == 0 {
+            length = default_length;
+        }
+
+        let additional = matches
+            .get_one::<String>(OPT_ADDITIONAL_SUFFIX)
+            .unwrap()
+            .to_string();
+        if additional.contains('/') || additional.contains('\\') {
```

Comment:
I think the check should probably be done with this function https://doc.rust-lang.org/std/path/fn.is_separator.html to be entirely correct, because only `/` should work on Unix and both `\` and `/` should work on Windows (and there might be other exceptions for other platforms).

It would probably look something like this:

```suggestion
        if additional.chars().any(std::path::is_separator) {
```

Mention me again when it's done and we'll merge this :)